src/librustc_trans/type_.rs

```diff
@@ -268,6 +269,20 @@ impl Type {
         }
     }
 
+    pub fn fat_ptr_data_type(&self) -> Type {
```

Comment:
Actually, `type_of::in_memory_type_of` isn't what I want. For `&Trait` it returns `{T*, (vtable)}` whereas I'm only trying to cast the base and thus want just `T*`.